### DISCUSSION TITLE
Select nested data keys from a dataset for plots

### DISCUSSION THREAD

**Discussion by captain-yoshi** [observablehq/plot#1050] [Q&A] [answered]
Here is my data:
![image](https://user-images.githubusercontent.com/32679594/190490304-fb05930a-dabc-40d1-90f1-c9a3dc9e9c55.png)

Is there a way to select a child or nested children like the example below ? Or do I have to modify my dataset to only have only one level.
```javascript
Plot.plot({
  marks: [
    Plot.boxY(data, {x: "query.request", y: "metrics.time"})
  ]
})
```

**Comment by mbostock** [comment, marked as answer]
Yes, you can use a function accessor: `{x: d => d.query.request, y: d => d.metrics.time}`. Though I think in your case you may need to flatten your data, as the `d.metrics.time` is an array of samples and Plot.boxY will expect a single number. That might look like this…

```js
Plot.plot({
  marks: [
    Plot.boxY(data.flatMap(d => d.metrics.time.map(t => ({request: d.query.request, time: t}))), {x: "request", y: "time"})
  ]
})
```

**Reply by captain-yoshi** [reply]
Super! Exactly what I wanted :)
